RecoBTag/Configuration/python/RecoBTag_cff.py

```diff
@@ -79,6 +79,10 @@
     
     # pixel cluster
     pixelClusterTagInfos,
+
+    # particle NET AK4 and AK8
+    pfParticleNetAK4TaskForRECO,
```

Comment:
Hi @rgerosa did you check what is the reco time increase?

Reply:
Hi @clacaputo 

just checked it as follows: 
1) First re-emulating first the HLT from a ttbar Run3 sample (GEN-SIM-DIGI-RAW), in order to produce a local file with 1k events. 
2) hen I re-run the DQM+RECO step on this file by either including or not the AK4 and AK8 PNET inferences (same cmsDriver commands described in https://twiki.cern.ch/twiki/bin/viewauth/CMS/HLTValidationAndDQM) . 

I am taking the overall timing and the module one from the process.option summary. I run on a local lxplus machine with 4 threads:

**With AK4 and AK8 PNET inference:**

TimeReport ---------- Event  Summary ---[sec]----
TimeReport       event loop CPU/event = 5.458949
TimeReport      event loop Real/event = 1.387473
TimeReport     sum Streams Real/event = 5.421264
TimeReport efficiency CPU/Real/thread = 0.983614

**Without AK4 and AK8 PNET inference:**

TimeReport ---------- Event  Summary ---[sec]----
TimeReport       event loop CPU/event = 5.405598
TimeReport      event loop Real/event = 1.372033
TimeReport     sum Streams Real/event = 5.367319
TimeReport efficiency CPU/Real/thread = 0.984961

Hence the difference is about 15ms per event that, I guess, is acceptable right? In particular because the ttbar MC should be a pessimistic scenario. Perhaps you need a cross check on some data events from single-mu or single-ele PD? TTbar is pessimistic because of the high occupancy of jets in the central region.

Finally in order to modify the b-tag sequence only for run3 era, is there a robust way to check the era append to the current process? Sorry for my ignorance, I was thinking to something that can be get on the fly via "hasattr" etc.

Reply:
> Finally in order to modify the b-tag sequence only for run3 era, is there a robust way to check the era append to the current process? Sorry for my ignorance, I was thinking to something that can be get on the fly via "hasattr" etc.

Usually, for sequence modification based on Eras, we use modifiers.

```python
from Configuration.Eras.Modifier_run3_common_cff import run3_common
_pfBTaggingTask_particleNet = pfBTaggingTask.copy()
_pfBTaggingTask_particleNet.add( pfParticleNetAK4TaskForRECO, pfParticleNetTask )
run3_common.toReplaceWith( pfBTaggingTask, _pfBTaggingTask_particleNet)
```

You and add it in [/Configuration/StandardSequences/python/Reconstruction_cff.py](https://github.com/cms-sw/cmssw/blob/master/Configuration/StandardSequences/python/Reconstruction_cff.py)

Reply:
@clacaputo thanks for the info! I have implemented it and tested successfully, therefore I am about to update the PR. One final thing: instead of inserting the customisation in the Reconstrution_cff I thought that, perhaps, it is cleaner to have it in the [RecoBTag/Configuration/python/RecoBTag_cff.py](https://github.com/cms-sw/cmssw/blob/master/RecoBTag/Configuration/python/RecoBTag_cff.py). In case you prefer instead to have it in the reconstruction file, please let me know!

Reply:
@rgerosa thanks for implementing it. Indeed, it makes sense to me